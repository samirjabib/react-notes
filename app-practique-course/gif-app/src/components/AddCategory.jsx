import { useState } from 'react';


export const AddCategdory = ({ onNewCategory }) => {

    const [ inputValue, setInputValue ] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue( target.value ); //Extraemos el valor del evento que esta pasando en nuestro navegador
    }

    const onSubmit = ( event ) => {

        event.preventDefault();/*
                                Para evitar el refresh del navegador
                                 todo al accionar el evento del form usamos este metodo.
                                */
        if( inputValue.trim().length <= 1) return; /*
                                                    Si ene el input no ahi mas de una letra, va a
                                                    romper la operacion ya que no cumple la condicion.
                                                    y con el return al lado nos ahorramos un scope adicional
                                                    al hacer la validacion.
                                                    
                                                    */

        // setCategories( categories => [ inputValue, ...categories ]);
        setInputValue('');
        onNewCategory( inputValue.trim() );
    }

    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}
