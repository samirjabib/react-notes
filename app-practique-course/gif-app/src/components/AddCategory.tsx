import { useInput } from "../hooks/useInput";
import { AddCategoryProps } from "../types/add-category.type";

export const AddCategory = ({ onNewCategory }: AddCategoryProps) => {
  const { inputValue, onInputChange, onSubmit } = useInput(onNewCategory);
  
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
