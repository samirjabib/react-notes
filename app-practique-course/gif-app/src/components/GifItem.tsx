


type GifItemProps = {
  title:string,
  url:string
}


export const GifItem = ({ title, url } : GifItemProps ) => {

  return (
    <div className="card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}
