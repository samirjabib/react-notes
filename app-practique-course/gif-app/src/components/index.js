
export * from './AddCategory'; //Exportamos todo lo que esta exportando nuestro archivo. 
export * from './GifGrid';
export * from './GifItem';