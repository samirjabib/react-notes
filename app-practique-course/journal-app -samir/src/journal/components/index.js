export * from './NavbarComponent';
export * from './SidebarComponent';
export * from './ImageGalleryComponent';