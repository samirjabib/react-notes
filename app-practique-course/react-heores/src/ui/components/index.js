

export * from './Navbar'; /*

                            Con esta funcion le indicamos que exporte todos  los modulos de la carpeta que 
                            le indicamos hacia el modulo superior.
                        */