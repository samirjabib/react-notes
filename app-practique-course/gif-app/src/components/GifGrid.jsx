import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
    /*
        Nota:
            No podemos usar la palabra reservada class
            se remplaza por className.
    
    */

    const { images, isLoading } = useFetchGifs( category ); /*Enviamos los datos mediante los argumentos del hook
                                                            que creamos empezando la cadena
                                                            y nos retorna los estados que necesitamos.
                                                            */
    
    return (
        <>
            <h3>{ category }</h3>
            {
                
                isLoading && ( <h2>Cargando...</h2> )
                                                    /*
                                                    
                                                        Usamos el operador && para mostrar algo mientras su condicion sea verdadera.
                                                     */
            }
            

            <div className="card-grid">
                {
                    images.map( ( image ) => (
                        <GifItem 
                            key={ image.id } 
                            { ...image }
                        />
                    ))
                }
                
            </div>

        </>
    )
}
