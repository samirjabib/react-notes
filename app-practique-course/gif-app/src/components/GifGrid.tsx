import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }: any) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}

      <div className="card-grid">
        {gifs.map((gif) => {
          const { id, title, url } = gif;
          return <GifItem key={id} title={title} url={url} />;
        })}
      </div>
    </>
  );
};
