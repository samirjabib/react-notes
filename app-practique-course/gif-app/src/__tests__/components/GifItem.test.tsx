import React from "react";

import { GifItem } from "../../components/GifItem";
import { render, screen } from "@testing-library/react";

describe("test in <GifItem/>", () => {
  const title = "saitama";
  const url = "htttps://one-puch.com/saitama.jpg";

  test("gif item should be render props", () => {
    const { container } = render(<GifItem title={title} url={url} />); //Extract container for compare this with the snapshot
    expect(container).toMatchSnapshot();
  });

  test("gif should be imagen with the exact url and alt", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug() //this is a console.log for component. 
    
    const {  } = screen.getByRole('img')
    c
    


  });
});