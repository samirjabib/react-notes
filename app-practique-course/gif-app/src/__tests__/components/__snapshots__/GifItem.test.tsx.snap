// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifItem/> gif item should be render props 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="htttps://one-puch.com/saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
